src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -396,19 +397,15 @@ public void execute(BuildScanExtension buildScan) {
             }
 
             if (isNotEmpty(gitRepo) && isNotEmpty(gitCommitId)) {
-                if (gitRepo.contains("github.com/") || gitRepo.contains("github.com:")) {
-                    Matcher matcher = Pattern.compile("(.*)github\\.com[/|:](.*)").matcher(gitRepo);
-                    if (matcher.matches()) {
-                        String rawRepoPath = matcher.group(2);
-                        String repoPath = rawRepoPath.endsWith(".git") ? rawRepoPath.substring(0, rawRepoPath.length() - 4) : rawRepoPath;
-                        buildScan.link("Github source", "https://github.com/" + repoPath + "/tree/" + gitCommitId);
-                    }
-                } else if (gitRepo.contains("gitlab.com/") || gitRepo.contains("gitlab.com:")) {
-                    Matcher matcher = Pattern.compile("(.*)gitlab\\.com[/|:](.*)").matcher(gitRepo);
-                    if (matcher.matches()) {
-                        String rawRepoPath = matcher.group(2);
-                        String repoPath = rawRepoPath.endsWith(".git") ? rawRepoPath.substring(0, rawRepoPath.length() - 4) : rawRepoPath;
-                        buildScan.link("GitLab Source", "https://gitlab.com/" + repoPath + "/-/commit/" + gitCommitId);
+                Matcher matcher = Pattern.compile("(https?://|git@)(.*?(github|gitlab).*?)[:|/](.*)").matcher(gitRepo);
+                if (matcher.matches()) {
+                    String repoUrl = "https://" + matcher.group(2) + "/";
+                    String rawRepoPath = matcher.group(4);
+                    String repoPath = rawRepoPath.endsWith(".git") ? rawRepoPath.substring(0, rawRepoPath.length() - 4) : rawRepoPath;
+                    if (gitRepo.contains("github")) {
+                        buildScan.link("Github source", repoUrl + repoPath + "/tree/" + gitCommitId);
+                    } else if (gitRepo.contains("gitlab")) {
+                        buildScan.link("GitLab Source", repoUrl + repoPath + "/-/commit/" + gitCommitId);
```

Comment:
Not sure I'm following. `gitHubUrl` is in another nested class `CaptureCiMetadataAction`. Do you mean to just go ahead and reference `GITHUB_SERVER_URL` and get the `repoUrl` from that if it is set?

Reply:
Just wanted to follow up on this @etiennestuder

Reply:
What I'm suggesting is that we also use gitHubUrl when creating the custom links to the source code. Right now, these custom links are only generated for gitlab.com and github.com, if I'm not mistaken. But, when hosting GitHub Enterprise at gh.foo.com, the source code links should also be created.

Or am I misunderstanding something here?

Reply:
So something like this...
```
Optional<String> gitHubUrl = envVariable("GITHUB_SERVER_URL", providers);
Optional<String> gitHubRepository = envVariable("GITHUB_REPOSITORY", providers);
if (gitHubUrl.isPresent() && gitHubRepository.isPresent()) {
    buildScan.link("GitHub source", gitHubUrl + "/" + gitHubRepository + "/tree/" + gitCommitId);
} else {
    // Generate build scan link to source using pattern matching
}
```
...would get you a build scan link to any enterprise github installation if running in GitHub Actions (or if those environment variables are otherwise set), no matter what that url is. Otherwise, just use the pattern matching already in this PR. Does that track with what you're thinking?

Reply:
`would get you a build scan link to any enterprise github installation if running in GitHub Actions ` --> yes, that's what I mean.

Reply:
Ok cool, sounds good. I'll get that updated.

Reply:
Done - the build scan link generation to github source has been updated following the example posted above using `GITHUB_SERVER_URL` and `GITHUB_REPOSITORY`